app/lib/shared/exceptions.dart

```diff
@@ -337,6 +337,20 @@ class OperationForbiddenException extends ResponseException {
         );
 }
 
+/// Thrown when the operation is rejected because a rate limit is reached.
+class RateLimitException extends ResponseException {
+  RateLimitException({
+    required int maxCount,
+    required String windowAsText,
+  }) : super._(
+          429,
+          'RateLimit',
+          'The operation is blocked, as rate limit in the current window '
+              'has been reached ($maxCount in the $windowAsText). '
+              'Please try again later.',
+        );
```

Comment:
We could opt to send a `Retry-After: $windowAsSeconds` header

Reply:
Done.